include/acl_thread.h

```diff
@@ -105,4 +116,39 @@ void acl_yield_lock_and_thread();
 } /* extern "C" */
 #endif
 
+class acl_wrapper_mutex {
```

Comment:
I think we should add some comments for the new class and Macro, e.g.: what's the purpose of this new class and etc.